src/ESM.sol

```diff
@@ -65,6 +74,34 @@ contract ESM {
         require(z >= x);
     }
 
+    // --- Auth ---
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+
+        emit Deny(usr);
+    }
+    modifier auth {
+        require(wards[msg.sender] == 1, "ESM/not-authorized");
+        _;
+    }
+
+    // -- admin --
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "min") {
+            require(data > 0, "ESM/min-required");
```

Comment:
Maybe `min > WAD` to cover magnitude errors too? Seems very unlikely the MKR supply will ever be so small a `min < WAD` would make sense.

Reply:
The year is 2208 and free-floating DAI has become the world reserve currency for centuries now. Many doubters over the years have caused above average burns going on for decades. The result is a mad-max style scramble for the few precious remaining MKR. A critical threshold will be reached in 6 months where the total supply drops below 1 WAD and the protocol will no longer be able to enforce ESM. The president of the world calls a national emergency to locate the great grandchildren of the fabled Protocol Engineering team and scold them for their grandparents not making better decisions.

Reply:
![Love it](https://media.giphy.com/media/xT0xeIbYVQcBFDSdVu/giphy.gif)